cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -323,27 +311,14 @@ class ARROW_EXPORT ExecNode {
   /// This may be called any number of times after StartProducing() succeeds.
   virtual void ResumeProducing(ExecNode* output, int32_t counter) = 0;
 
-  /// \brief Stop producing definitively to a single output
-  ///
-  /// This call is a hint that an output node has completed and is not willing
-  /// to receive any further data.
-  virtual void StopProducing(ExecNode* output) = 0;
-
-  /// \brief Stop producing definitively to all outputs
-  virtual void StopProducing() = 0;
-
-  /// \brief A future which will be marked finished when this node has stopped producing.
-  virtual Future<> finished() { return finished_; }
+  /// \brief Abort execution and perform any needed cleanup (such as closing files, etc.)
```

Comment:
What does `Abort execution` mean for a node?  In theory all "execution" is handled via the scheduler so does a node really need to do anything here?  Why `ExecNode::Abort` instead of doing the cleanup in the `ExecNode` destructor?